includes/Modules/Analytics.php

```diff
@@ -133,10 +105,87 @@ function () {
 			0
 		);
 
+		// Analytics tag placement logic.
 		add_action(
-			'admin_init',
+			'template_redirect',
 			function() {
-				$this->handle_provisioning_callback();
+				// Bail early if we are checking for the tag presence from the back end.
+				if ( $this->context->input()->filter( INPUT_GET, 'tagverify', FILTER_VALIDATE_BOOLEAN ) ) {
+					return;
+				}
+
+				$use_snippet = $this->get_data( 'use-snippet' );
+				if ( is_wp_error( $use_snippet ) || ! $use_snippet ) {
+					return;
+				}
+
+				$property_id = $this->get_data( 'property-id' );
+				if ( is_wp_error( $property_id ) || ! $property_id ) {
+					return;
+				}
+
+				// At this point, we know the tag should be rendered, so let's take care of it
+				// for AMP and non-AMP.
+				if ( $this->context->is_amp() ) {
+					$print_amp_gtag = function() use ( $property_id ) {
+						// This hook is only available in AMP plugin version >=1.3, so if it
+						// has already completed, do nothing.
```

Comment:
This is a bit off to read since this callback is used for multiple hooks now.

Reply:
Good point, pushed another commit to change these.

Reply:
The only reason I think something like that would be necessary is if the AMP plugin printed the GA snippet itself and we needed to check to see if this action was already done to prevent duplicate output, but it looks like the output is only our own. So if that's the case, then this is less verbose and more consistent with the rest 👍